src/VoiceConnection.ts

```diff
@@ -248,17 +258,22 @@ export class VoiceConnection extends TypedEmitter<VoiceConnectionEvents> {
 		const oldSubscription: PlayerSubscription | undefined = Reflect.get(oldState, 'subscription');
 		const newSubscription: PlayerSubscription | undefined = Reflect.get(newState, 'subscription');
 
-		if (oldNetworking && oldNetworking !== newNetworking) {
-			oldNetworking.off('debug', this.onNetworkingDebug);
-			oldNetworking.on('error', noop);
-			oldNetworking.off('error', this.onNetworkingError);
-			oldNetworking.off('close', this.onNetworkingClose);
-			oldNetworking.off('stateChange', this.onNetworkingStateChange);
-			oldNetworking.destroy();
+		if (oldNetworking !== newNetworking) {
+			oldNetworking?.off('debug', this.onNetworkingDebug);
+			oldNetworking?.on('error', noop);
```

Comment:
Is this intentional?

Reply:
Also, I'd do a nested if here for all those `off`s, otherwise we're running the same logic 6 times, which is not optimal.